ballerina/xmldata.bal

```diff
@@ -43,53 +43,63 @@ public type JsonOptions record {
 # + return - XML representation of the given JSON if the JSON is
 # successfully converted or else an `xmldata:Error`
 public isolated function fromJson(json jsonValue, JsonOptions options = {}) returns xml?|Error {
+    map<string> allNamespaces = {};
     if !isSingleNode(jsonValue) {
-        return getElement("root", check traverseNode(jsonValue, {}, options),
-                           check getAttributesMap(jsonValue, options = options));
+        addNamespaces(allNamespaces, check getNamespacesMap(jsonValue, {}, options));
+        return getElement("root", check traverseNode(jsonValue, allNamespaces, {}, options), allNamespaces,
+                            check getAttributesMap(jsonValue, allNamespaces, options = options));
     } else {
         map<json>|error jMap = jsonValue.ensureType();
         if jMap is map<json> {
             if jMap.length() == 0 {
                 return xml ``;
             }
             json value = jMap.toArray()[0];
+            addNamespaces(allNamespaces, check getNamespacesMap(value, {}, options));
             if value is json[] {
-                return getElement("root", check traverseNode(value, {}, options, jMap.keys()[0]),
-                                check getAttributesMap(value, options = options));
+                return getElement("root", check traverseNode(value, allNamespaces, {}, options, jMap.keys()[0]),
+                                allNamespaces, check getAttributesMap(value, allNamespaces, options = options));
             } else {
                 string key = jMap.keys()[0];
                 if key == CONTENT {
                     return xml:createText(value.toString());
                 }
-                return getElement(jMap.keys()[0], check traverseNode(value, {}, options),
-                                check getAttributesMap(value, options = options));
+                addNamespaces(allNamespaces, check getNamespacesMap(value, {}, options));
```

Comment:
Do we need to call `addNamespaces` again inside else block?

Reply:
Removed now